pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2191,10 +2198,19 @@ func (a *AWSCloud) getInstancesByNodeNames(nodeNames []string) ([]*ec2.Instance,
 // Returns the instance with the specified node name
 // Returns nil if it does not exist
 func (a *AWSCloud) findInstanceByNodeName(nodeName string) (*ec2.Instance, error) {
+	var filterKey string
+	if a.cfg.Global.UseKubernetesNodeTag {
+		filterKey = "tag:" + TagNameKubernetesNode
+	} else {
+		filterKey = "private-dns-name"
+	}
+
 	filters := []*ec2.Filter{
-		newEc2Filter("private-dns-name", nodeName),
+		newEc2Filter(filterKey, nodeName),
+		newEc2Filter("instance-state-name", "running"),
```

Comment:
I think adding this is-running filter here is probably a good thing, but it is a behavioural change.  Can we split it out into its own PR.  Or is it important to have in this PR for some reason?